linera-execution/src/runtime.rs

```diff
@@ -33,12 +33,22 @@ use std::{
 /// Runtime data tracked during the execution of a transaction.
 #[derive(Debug, Clone)]
 pub(crate) struct ExecutionRuntime<'a, C, const WRITABLE: bool> {
+    /// The amount of fuel available for executing the application.
+    remaining_fuel: Arc<AtomicU64>,
+    /// the number of read
+    n_read: Arc<AtomicU64>,
+    /// the total size being read
+    bytes_read: Arc<AtomicU64>,
+    /// the total size being write
+    bytes_write: Arc<AtomicU64>,
+    /// The maximum size of read allowed
+    maximum_bytes_read: u64,
+    /// The maximum size of write allowed
+    maximum_bytes_write: u64,
```

Comment:
Why not `Arc<RuntimeMeter>` where `RuntimeMeter` uses atomic integers?
Then you attach a number of (`&self`) update methods to `RuntimeMeter`

Reply:
I did change the structure of the code and added an `update_limits` method to the `RuntimeLimits`.
But the structure has changed a lot now.